fortex/health/processors/scispacy_processor.py

```diff
@@ -0,0 +1,170 @@
+        self.set_up()
+
+    def _process(self, input_pack: DataPack):
+        r"""
+        ScispaCyProcessor is done on the basis of
+        using SciSpacy and the corresponding
+        trained model for Hyponym, Abbreviation
+
+        """
+
+        path_str, module_str = self.configs.entry_type.rsplit(".", 1)
+
+        mod = importlib.import_module(path_str)
+        entry = getattr(mod, module_str)
+        for entry_specified in input_pack.get(entry_type=entry):
+
+            doc = self.extractor(entry_specified.text)
+
+            if self.configs.pipe_name == "abbreviation_detector":
+                # print("Abbreviation", "\t", "Definition")
+                list_of_abrvs = []
+                for abrv in doc._.abbreviations:
+                    print(
+                        f"{abrv} \t ({abrv.start}, {abrv.end}) {abrv._.long_form}"
+                    )
+                    tmpAbrv = Abbreviation(
+                        pack=input_pack, begin=abrv.start, end=abrv.end
+                    )
+                    tmpAbrv.long_form = abrv._.long_form
+                    list_of_abrvs.append(tmpAbrv)
+
+            else:
+                for item in doc._.hearst_patterns:
+                    general_concept: Phrase = Phrase(
+                        pack=input_pack, begin=item[1].start, end=item[1].end
+                    )
+                    specific_concept = Phrase(
+                        pack=input_pack, begin=item[2].start, end=item[2].end
+                    )
+                    hlink = Hyponym(
+                        pack=input_pack,
+                        parent=general_concept,
+                        child=specific_concept,
+                    )
+                    hlink.hyponym_link = item[0]
+
+    @classmethod
+    def default_configs(cls):
+        r"""
+        This defines a basic config structure for `ICDCodingProcessor`.
+
+        Following are the keys for this dictionary:
+         - `entry_type`: input entry type,
+         - `model_name`: the higgingface transformer model name to be
+                         used for classification,
+
+        Returns: A dictionary with the default config for this processor.
+        """
+        return {
+            "entry_type": "ft.onto.base_ontology.Document",
+            "attribute_name": "classification",
+            "multi_class": True,
+            "model_name": "en_core_sci_sm",
+            "pipe_name": "abbreviation_detector",
+            "cuda_devices": -1,
+        }
+
+    def expected_types_and_attributes(self):
+        r"""
+        Method to add user specified expected type which would be checked
+        before running the processor if the pipeline is initialized with
+        `enforce_consistency=True` or
+        :meth:`~forte.pipeline.Pipeline.enforce_consistency` was enabled for
+        the pipeline.
+        """
+        return {
+            "ft.onto.base_ontology.Document": set(),
```

Comment:
expected type is not `ft.onto.base_ontology.Document`, but should be `self.configs.entry_type`

Reply:
Ok this line is changed to : self.configs.entry_type: set(),   instead